src/Npgsql/NpgsqlCommand.cs

```diff
@@ -515,7 +517,7 @@ void DeriveParametersForQuery(NpgsqlConnector connector)
                         }
                     }
 
-                    var msg = connector.ReadMessageWithoutCancellation();
+                    var msg = connector.ReadMessage(async: false, pgCancellation: false).GetAwaiter().GetResult();
```

Comment:
Yes, the sync API is now ugly - but that's a feature :) If we care enough, we'll have to properly implement async for this :rofl: